src/main/java/frc/robot/subsystems/Led.java

```diff
@@ -17,26 +19,88 @@
     private int frame = 0;
     private int timer = 0;
 
+    public enum BallColor {
+        RED(Color.kFirstRed), BLUE(Color.kFirstBlue), NONE(new Color(0, 0, 0));
+
+        private final Color color;
+
+        private BallColor(Color color) {
+            this.color = color;
+        }
+
+        public Color get() {
+            return color;
+        }
+    }
+
     public Led(final LedMap map) {
         led = map.getLed();
         ledBuffer = map.getLedBuffer();
         led.setLength(ledBuffer.getLength());
         led.start();
     }
 
+    private void runAnimation(LightAnimation animation, double brightness) {
+        if (timer % 5 == 0) {
+            for (int i = 0; i < ledBuffer.getLength(); i++) {
+                Color c = animation.getColor(frame,
+                        i / (ledBuffer.getLength() / 10));
+                ledBuffer.setLED(i, new Color(
+                        c.red * brightness,
+                        c.green * brightness,
+                        c.blue * brightness));
+            }
+            led.setData(ledBuffer);
+            frame++;
+        }
+    }
+
     public CommandBase animate(LightAnimation animation, double brightness) {
         return running("Animate", () -> {
-            if (timer % 5 == 0) {
+            runAnimation(animation, brightness);
+            timer++;
```

Comment:
@bot190 Is this a pattern that we want to use? And encourage? Or is there something in wpilib that I'm missing that does this for us?

If we're going to suggest this, it might be beneficial to make a wrapper for `everyNCycles` in chopshoplib...